api/proto/teleport/autoupdate/v1/autoupdate.proto

```diff
@@ -33,8 +33,25 @@ message AutoUpdateConfig {
 
 // AutoUpdateConfigSpec encodes the parameters of the autoupdate config object.
 message AutoUpdateConfigSpec {
-  // ToolsAutoupdate encodes the feature flag to enable/disable tools autoupdates.
-  bool tools_autoupdate = 1;
+  reserved 1;
+  reserved "tools_autoupdate"; // ToolsAutoupdate is replaced by tools.mode.
+  AutoUpdateConfigSpecTools tools = 2;
+}
+
+// AutoUpdateConfigSpecTools encodes the parameters for client tools auto updates.
+message AutoUpdateConfigSpecTools {
+  // Mode defines state of the client tools auto update.
+  ToolsUpdateMode mode = 1;
+}
+
+// ToolsMode type for client tools enabling state.
+enum ToolsUpdateMode {
+  // TOOLS_UPDATE_MODE_UNSPECIFIED is undefined mode.
+  TOOLS_UPDATE_MODE_UNSPECIFIED = 0;
+  // TOOLS_UPDATE_MODE_ENABLED client tools auto update is enabled.
+  TOOLS_UPDATE_MODE_ENABLED = 1;
+  // TOOLS_UPDATE_MODE_DISABLED client tools auto update is disabled.
+  TOOLS_UPDATE_MODE_DISABLED = 2;
```

Comment:
Can you make the mode a string instead of an enum? Sorry for the last minute change, it appears I made a mistake in the RFD and this should have been a string.

There is nothing wrong technically with it being an enum (technically it's better as we have stricter content validation), however we use the protobuf messages to generate Kubernetes CRDs and Terraform Schemas. Seeing integers instead of strings is not user friendly as it's not clear that `mode: 2` means "updates are disabled".

I will update RFD 184 to fix the protos.